magicgui/widgets/_bases/widget.py

```diff
@@ -66,6 +67,7 @@ def __init__(
         visible: bool | None = None,
         enabled: bool = True,
         gui_only=False,
+        parent: Any = None,
```

Comment:
The `parent` is not documented in the docstring. As this is a fragile part of this PR. There should be informed that if the parent widget is removed (for example, Qt native one) then the object will be unavailable, and trying to access it may end with `RuntimeError: C++ wrapped object has been destroyed.` 

I think that it is fragile as many magicgui users are not familiar with Qt and its errors.

Reply:
added docs,  it's too much to tackle in this PR.  An approval would be appreciated so I can merge (without having to force it)